pkg/storage/ingest/version.go

```diff
@@ -17,6 +17,56 @@ const (
 	V2RecordSymbolOffset   = 64
 )
 
+var (
+	// Bake in a shared symbols table for extremely common symbols.
+	// The index corresponds to the symbol value.
+	// The first `V2RecordSymbolOffset` symbols are reserved for this table.
+	// Note: V2 is not yet stabilized.
+	V2CommonSymbols = []string{
+		// Prometheus/Mimir symbols
+		"__name__",
+		"__aggregation__",
+		"<aggregated>",
+		"le",
+		"component",
+		"cortex_request_duration_seconds_bucket",
```

Comment:
I am surprised to see that this is more frequent than `pod`. Is this sorted by frequency? I didn't check how do we encode the references to the symbol table, but I guess they're varints and we should aim for a Huffman-like coding here.